client/src/components/Panels/ToolBox.vue

```diff
@@ -28,10 +28,10 @@
             <div class="toolMenuContainer">
                 <div class="toolMenu">
                     <tool-section
-                        v-for="section in sections"
+                        v-for="(section, key) in sections"
                         :category="section"
                         :query-filter="queryFilter"
-                        :key="section.id"
+                        :key="key"
                         @onClick="onOpen"
```

Comment:
You might want to adopt a consistent event/handler naming convention. Typically the handler will be named "onSomething" and the event will be named "something"